### DISCUSSION TITLE
The temperature of the DS18B20 probes on FGK101 no longer rises

### DISCUSSION THREAD

**Discussion by OUARZA** [zwave-js/zwave-js#6439] [Request Support / Investigate Issue]
### Checklist

- [X] I have read and followed the above instructions
- [X] I have checked the [troubleshooting section](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index) and my problem is **not** described there.

- [X] I have read the [changelog](https://github.com/zwave-js/node-zwave-js/blob/master/CHANGELOG.md) and my problem was **not** mentioned there or the fix **did not work**.

### Describe the issue

I use ZWaveJS through Jeedom.
Since version 8.25.1 of ZwaveJsUI the temperature of the DS18B20 probes of my FGK101 equipment no longer rises.

![image](https://github.com/zwave-js/node-zwave-js/assets/34892335/0d1241c2-f3b6-489a-87c1-5e1eaeb999f5)

### Software versions

Driver (node-zwave-js): ...

Z-Wave JS UI:
zwave-js-ui: 8.26.0.08e6a7e
zwave-js: 11.14.3
home id: 3972900806
home hex: 0xeccda7c6

Home Assistant Z-Wave Integration: ...

Home Assistant Z-Wave JS Addon: JEEDOM

ioBroker.zwave2 Adapter: ...

If you are using something non-standard, tell us here: ...

### Device information

"productLabel" : "FGK101",
"productDescription" : "Door Opening Sensor",
"manufacturer" : "Fibargroup",
"firmwareVersion" : "2.1",
NodeID : 35

### Checklist

- [X] I [made sure](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index?id=rule-1-use-the-driver-log-luke) to provide a **driver log** on level **debug**.

- [X] The log includes a re-interview of the problematic device _(if applicable)_.

- [X] The log includes the problematic interaction with the device _(if applicable)_.

- [X] I provided the node ID of the problematic device _(if applicable)_.

### Upload Logfile

[zwave-js-ui-store.zip](https://github.com/zwave-js/node-zwave-js/files/13062821/zwave-js-ui-store.zip)

[node_35.json](https://github.com/zwave-js/node-zwave-js/files/13062827/node_35.json)

[nodes_dump.json](https://github.com/zwave-js/node-zwave-js/files/13062832/nodes_dump.json)

The screenshot in the help is no longer good.
https://zwave-js.github.io/zwave-js-ui/#/troubleshooting/generating-logs?id=driver-logs
![image](https://github.com/zwave-js/node-zwave-js/assets/34892335/25aa2543-a2ad-44f1-9201-b571dea6296b)

**Comment by AlCalzone** [comment]
There were no recent changes that would explain this, but in your dump the endpoint 2 where the sensor is exposed is missing. Please re-interview the device, make another driver log of that and upload it here again (just the current log).

**Comment by OUARZA** [comment]
@AlCalzone 
Okay, I'll do that.
I have the problem with version 8.26.0
Is it possible to have the JSON file of version 8.22.0?
https://github.com/zwave-js/node-zwave-js/blob/master/packages/config/config/devices/0x010f/fgk101.json

**Reply by AlCalzone** [reply]
Not sure what's that supposed to accomplish. The last change was in v10.9 (Z-Wave JS UI 8.13.1) and that was meant to **fix** the missing sensor.

**Comment by OUARZA** [comment]
@AlCalzone Ok, I have the impression that the problem comes from this file, but not sure.
Is the log file you want after waking up the module? zwave-js-ui-store.zip

**Reply by AlCalzone** [reply]
The logfile I need is `zwavejs_<date>.log` or `zwavejs_current.log`. Not sure if you selected that one, because there's nothing uploaded.

**Reply by OUARZA** [reply]
@AlCalzone What folder do I find it in?

**Reply by OUARZA** [reply]
[zwavejs_current.log](https://github.com/zwave-js/node-zwave-js/files/13068801/zwavejs_current.log)
[zwavejs_2023-10-23.log](https://github.com/zwave-js/node-zwave-js/files/13068802/zwavejs_2023-10-23.log)

**Reply by AlCalzone** [reply]
Those files are empty

**Reply by OUARZA** [reply]
You can verify ?

**Reply by AlCalzone** [reply]
There's no interview in the logfile. If this is a battery-powered device you may have to wake it up manually.

**Reply by OUARZA** [reply]
However, that’s what I did. I woke him up manually. The date and time have changed.
If in doubt, I will give you the 2 files I made now.
![image](https://github.com/zwave-js/node-zwave-js/assets/34892335/f8d089db-f47b-4999-8c7e-a3a02f7cb9ec)

[zwave-js-ui-store.zip](https://github.com/zwave-js/node-zwave-js/files/13069046/zwave-js-ui-store.zip)

**Reply by AlCalzone** [reply]
Ok, let's do this again:
1. Make sure logging to file is still enabled and the loglevel is on debug
2. Click Re-Interview for the device
3. Then wake it up, make sure it's waking up in the UI

The "Complete" status on the right should now go through the different interview stages. When it's back on Complete, send me the log.

**Reply by OUARZA** [reply]
There you go, I woke him up several times.
I would like to point out that I have the problem on all of my FGK101 which have a probe.
![image](https://github.com/zwave-js/node-zwave-js/assets/34892335/b6ba5995-fdd3-49e5-9954-00514be45530)

[zwave-js-ui-store.zip](https://github.com/zwave-js/node-zwave-js/files/13069602/zwave-js-ui-store.zip)

**Reply by AlCalzone** [reply]
Hm, this may be a bug in the driver. The device does not send an actual wakeup notification, but a different command when it wakes up. So the interview does not continue.

Aside from that:
According to the [manual](https://manuals.fibaro.com/content/manuals/en/FGK-10x/FGK-10x-EN-A-v1.00.pdf), the device is supposed to have associations where it reports the current status. Looking at your log however, Z-Wave JS polls every time the device wakes up, so I think the lifeline association is missing.
Please check the "groups" tab if there is node 1 assigned to group 3.

Did you connect the temperature sensor **before** including the device in the network? If not, the device won't tell us about the sensor.

**Reply by AlCalzone** [reply]
FWIW, I've tried to reproduce the re-interview issue, but I can't. When I schedule a re-interview on a sleeping device and then have it send the NIF like yours does, the re-interview does happen.

**Reply by OUARZA** [reply]
It has always worked very well.
I have had the problem on all of my FGK101s since 09/15 when switching to 8.25.1.
When I associated it, a long time ago, the probe was there before making the association.

![image](https://github.com/zwave-js/node-zwave-js/assets/34892335/7c12e9b9-3b5b-48bf-b3cd-2f467daaaf9e)

**Reply by AlCalzone** [reply]
Ok so the association is there. Wonder why it doesn't report on its own then.
Any chance you could exclude one of the devices and include it again while still making a log?

**Reply by OUARZA** [reply]
I'll see if I have another one in the cupboard with a probe.

**Reply by AlCalzone** [reply]
I went through the entire change history between 8.22.0 and 8.25.1. The only change I can see that might be related is

> Discard Meter CC and Multilevel Sensor CC reports when the node they supposedly come from does not support them (#6178)

which is included in 8.23.1.

This was actually broken for a while if the report is from an endpoint and the node does not support the CC (which is likely the case here), but this fix
> Fixed a regression from v11.13.0 where Meter CC and Multilevel Sensor CC reports from an endpoint were discarded, although the endpoint did support them, but the root endpoint didn't (#6222)

is included in 8.24.1. This only leaves Z-Wave JS not knowing about endpoint 2 for whatever reason, which would explain what you're seeing. But this means the endpoint was likely never known, probably because you connected the temperature sensor after joining the network. 
This is unfortunately how it needs to be done in Z-Wave, but the manual also explains it:
![grafik](https://github.com/zwave-js/node-zwave-js/assets/17641229/ef73c1aa-aa3a-4907-a407-1e753aba8339)

Or you joined it before v10.9 (Z-Wave JS UI 8.13.1) and never re-interviewed to pick up that fix. I still don't understand why the re-interview doesn't start for you...

**Reply by OUARZA** [reply]
The inclusion was done successfully with the probe, but there was a moment.
What I can do is exclude the module without deleting it (to avoid losing my history) and I re-include it. What do you think ?

**Reply by AlCalzone** [reply]
One thing that can be done to replace a node without changing the ID is:

- stop Z-Wave JS UI
- factory reset the device
- start Z-Wave JS UI
- Use "replace failed node" to replace the device with itself

---

Not sure how your system handles this, so **I'd try something less destructive first:**
Triggering a re-interview by default waits for the device to wake up, so it stays ready to use until it actually wakes up and the re-interview can be performed. For whatever reason this doesn't seem to work for you, so you can also trigger a re-interview without waiting manually:
**Menu on the bottom right -> Advanced -> Driver function -> Insert this code**
```ts
const node = driver.controller.nodes.get(  35  );
node.refreshInfo({ waitForWakeup: false });
```

This will immediately treat your node as if you had just included it. You will have to wake it up for Z-Wave JS to interview it and make it usable again. If nothing happens, you may have to restart Z-Wave JS UI first.

Do this for each of the sensors, every time replace the 35 in the first line with the current sensor's node ID

**Reply by OUARZA** [reply]
@AlCalzone 
here ?
![image](https://github.com/zwave-js/node-zwave-js/assets/34892335/34cf3de5-7de7-4d74-8bf2-f335004bea9e)

**Reply by AlCalzone** [reply]
Yes

**Reply by OUARZA** [reply]
@AlCalzone 
I just did the manipulation and it works again!
I would like to point out that other people on the JEEDOM forum have the same problem as me.

![image](https://github.com/zwave-js/node-zwave-js/assets/34892335/969fea4f-79df-4e61-98dc-cef917341e93)

![image](https://github.com/zwave-js/node-zwave-js/assets/34892335/7b2b7d11-ca03-406b-baa4-dd3ff424b21a)

**Reply by AlCalzone** [reply]
> other people on the JEEDOM forum have the same problem

Do you know if re-interviewing actually does something for them?
As said before, I can't reproduce the issue with re-interview not starting in tests, but I also have no old enough device that behaves like this one that I could use to do a manual test.

**Reply by OUARZA** [reply]
I will ask them :)

**Reply by OUARZA** [reply]
Hello,
I'm sorry, the people with the problem didn't answer me :(

**Comment by Overbull** [comment]
Hello, I did a re-interviewing as @AlCalzone suggested on my 7 FGK101 and the temperature rises again on all of them, even a correct battery level. 

Try to do from zwave-js control panel not from jeedom.
